query/common/hll.go

```diff
@@ -688,11 +697,33 @@ func (hll *HLL) Encode() []byte {
 	if len(hll.DenseData) != 0 {
 		return hll.DenseData
 	}
-	data := make([]byte, 3*len(hll.SparseData))
+	return hll.encodeSparse(false)
+}
+
+// EncodeBinary converts HLL to binary format
+// aligns to 4 bytes for sparse hll
+// used to build response for application/hll queries from HLL struct
+func (hll *HLL) EncodeBinary() []byte {
+	if len(hll.DenseData) != 0 {
+		return hll.DenseData
+	}
+	return hll.encodeSparse(true)
+}
+
+func (hll *HLL) encodeSparse(padding bool) []byte {
```

Comment:
let's maybe add some comment about the two ways of encoding format
1. padding (1 byte) | rho (1 byte) | index (2 byte)   endianness based on the machine itself
2.                                rho (1 byte) | index (2 byte)  small endian